### ISSUE TITLE
Ble Adapter aktualisiert Messwerte nach 18 - 30 h nicht mehr

### ISSUE THREAD

**Issue by Marty56** [AlCalzone/ioBroker.ble#391]
Ich habe einen Xiaomi Flora Care. Auf meinem Raspberry 3 habe ich den BLE Adapter installiert. Der BLE Adapter hat den Flora Care  sofort  erkannt, hört aber nach 18-30 Std auf, die Werte vom Sensor zu aktualisieren.
Wenn ich den BLE Adapter neu starte, funktioniert es wieder.
Nach Neustart des Raspberries werden die Werte auch erst aktualisiert, wenn ich den BLE Adapter manuell neu starte.

Im Log gibt es keine Fehler oder Hinweise. Der BLE Adapter steht auf grün und connection auf true und driverState auf powerON.

**Comment by AlCalzone** [comment]
Um hier irgendwas erkennen zu können, benötige ich einen Log mit Loglevel "debug", in dem dieser Zeitraum abgedeckt ist (Achtung, kann recht groß werden!).

**Comment by Marty56** [comment]
Ich habe 'debug' probehalber eingestellt. Allerdings werden so viele Daten erzeugt, dass ich vermute, dass die SSD meines Raspberries nicht ausreichend Kapazität dafür hat. Das ist also nicht praktikabel. Ich müsste  also das log auf mein NAS umleiten. Vielleicht mache ich das irgendwann mal.

Ich habe jetzt einen Workaround, dass ich per Script die Änderung des RSSI Werts überprüfe und wenn der RSSI Wert sich nicht innerhalb von 10 Minuten upgedatet hat, dann starte ich - auch per Script - den Adapter neu.

**Comment by Marty56** [comment]
Ich habe die Umleitung auf mein NAS gemacht. Es werden pro Minute knapp 1 MB log Daten erzeugt. Bei 24 h würde ich als 1,4 GB log Daten erwarten.
Sollte kein Problem für mein Nas sein.
See you in 24 h.

**Comment by AlCalzone** [comment]
Ich habe mal bei den Maintainern von `noble` nachgefragt, da ich vermute, dass der Fehler dort liegt.

**Comment by harrym67** [comment]
ich häng mich hier mal dazu.
Ich hab ne MiBand 4 die auch beim ersten Start erkannt wird und RSSI daten liefert. Allerdings nur einmal. Im Log sehe ich, dass der BLE Adapter den Scan aktiviert und wieder deaktiviert. Die RSSI Daten vom MiBand werden aber nicht aktualisiert.

Stoppen vom Adapter, 10 min warten, Adapter neu starten ..... Daten kommen sofort wieder.

**Comment by AlCalzone** [comment]
> und wieder deaktiviert

Woran erkennst du das?

**Comment by Marty56** [comment]
> Ich habe mal bei den Maintainern von `noble` nachgefragt, da ich vermute, dass der Fehler dort liegt. Ich glaube weitere Daten brauche ich nicht, dein kurzer Log zeigt das Problem schon.

Hmmm .. wird 'nobel' überhaupt noch gepflegt?  Die letzten Änderung in Git sind von 2018.

**Comment by mrsmarthome** [comment]
Hallo, ich hatte/habe ein vergleichbares Problem:
ich habe am 02.01.2021 angefangen mein RaspberryPi 4 mit den Aufgaben der Heizungssteuerung zu versehen, habe aber um ehrlich zu sein wenig bis keine Ahnung vom Programmieren bzw. von Raspbian. Ich konnte bis gestern Abend noch meine Heizungssteuerung laufen lassen, heute morgen ist mir dann aufgefallen, dass der BLE Adapter keine aktuellen Werte mehr einliest, obwohl die Geräte in der Nähe sind und funktionieren.

Zu meiner Schande muss ich zugeben, dass ich gestern oder vorgestern diverse Packages und Adapter geupdatet habe, warum weiß ich um ehrlich zu sein nicht.

Leider weiß ich auch nicht genau, wie ich ein Debug Log erstelle bzw. wo ich ein entsprechendes finde, damit ich es mir entweder mal selbst ansehen könnte oder aber @AlCalzone zur Verfügung stellen könnte.

Kann mir hier jemand helfen?

**Comment by AlCalzone** [comment]
Du kannst bei aktiviertem Expertenmodus in der Instanzen-Liste das Loglevel ändern

**Comment by mrsmarthome** [comment]
okay, das habe ich jetzt gemacht, mir ist das nichts aufgefallen. Vielleicht könntest Du @AlCalzone mal reinsehen? 
[iobroker.2021-01-12.log.txt](https://github.com/AlCalzone/ioBroker.ble/files/5802476/iobroker.2021-01-12.log.txt)

**Comment by AlCalzone** [comment]
```
2021-01-12 14:49:15.354 - debug: ble.0 (4136) xiaomi >> got data: 50305b05fa2776a938c1a40d1004b7001a02
2021-01-12 14:49:15.355 - debug: ble.0 (4136) xiaomi >> got humidity update => 53.8
2021-01-12 14:49:15.356 - debug: ble.0 (4136) xiaomi >> got temperature update => 18.3
2021-01-12 14:49:15.358 - debug: ble.0 (4136) a4:c1:38:a9:76:27 > got values: {"humidity":53.8,"temperature":18.3}
2021-01-12 14:49:15.358 - debug: ble.0 (4136) setting state ble.0.a4:c1:38:a9:76:27.humidity
2021-01-12 14:49:15.363 - debug: ble.0 (4136) setting state ble.0.a4:c1:38:a9:76:27.temperature
```
Den logs nach zu urteilen kommt da was rein

**Comment by mrsmarthome** [comment]
Ja, aber nur bei diesem Objekt (das ist ein Xiami Temperatur/Feuchtigkeitssensor der zweiten Generation. Die anderen sehen folgendermaßen aus
![übersicht über objekte](https://user-images.githubusercontent.com/77334947/104327914-92179980-54eb-11eb-8154-82750f4505c6.JPG)

**Comment by AlCalzone** [comment]
Joar, von denen wird rein gar nichts empfangen. Könnte vieles sein... Empfangsprobleme, ein anderer Adapter (RADAR), der ggf. dazwischen funkt, eine [fehlerhafte bluetooth-Firmware](https://github.com/RPi-Distro/firmware-nonfree/issues/8) des Pi 4

**Comment by mrsmarthome** [comment]
aber das hatte gestern noch funktioniert und der Sensor ist ca. 10cm entfernt, also daran sollte es nicht liegen...

**Comment by AlCalzone** [comment]
🤷🏻‍♂️

**Comment by mrsmarthome** [comment]
okay, ich hatte mich gestern Abend wohl zu früh gefreut. Der Adapter erkennt jetzt zumindest den anderen Sensor, aktualisiert jedoch nur die Einträge rssi und services, die "wichtigen" Daten bleiben immer noch auf dem alten Wert bestehen. Ich habe auch ein neues debug Logfile erstellt, vielleicht hilft es Dir weiter:
[iobroker.2021-01-13.log.txt](https://github.com/AlCalzone/ioBroker.ble/files/5807199/iobroker.2021-01-13.log.txt)

im angefügten Screenshot ist mir auch noch aufgefallen, dass im Vergleich zu dem Sensor, dessen Werte aktualisiert werden, die Werte des nicht aktualisierten Sensors von einem anderen "user" geschrieben werden:
![übersicht user](https://user-images.githubusercontent.com/77334947/104423158-0733af00-557e-11eb-9015-c9b689abc8be.JPG)

**Comment by AlCalzone** [comment]
Laufen die ggf. unter einer custom firmware?

**Comment by mrsmarthome** [comment]
ja, also die "neuen" Sensoren sind eckig (Modellname: LYWSD03MMC) und die Daten sind verschlüsselt. Man kann sie über die XiaomiHome App oder so aktivieren, aber ich glaube, dass man die Daten dann auch nur in der App sieht und die Sensoren ihre Daten immer noch verschlüsselt versenden.

Ich habe auf Youtube eine Anleitung (hier: [https://www.youtube.com/watch?v=NXKzFG61lNs](https://www.youtube.com/watch?v=NXKzFG61lNs ))gefunden, mit der man die Dinger encrypten kann, das habe ich auch bei beiden Sensoren gemacht. Der Sensor, der jetzt noch Werte liefert, hatte jedoch irgendwie Probleme beim flashen gemacht, aber nach ein paar mal neustarten, hat er Werte an BLE übertragen.
